app/controllers/api/v1/current_user_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class CurrentUserController < ApiBaseController
+      # collection of API methods related to the current user object.
+      # for security reasons, the methods in this controller should be extremely tightly controlled
+      include Concerns::Authenticator
+      include Concerns::StudyAware
+      include Swagger::Blocks
+
+      before_action :set_current_api_user!
```

Comment:
You want `:authenticate_api_user!` instead of `:set_current_api_user!` - this automatically returns `401` and halts execution.